src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -277,12 +315,13 @@ def _add_input_args(parser):
 
     prompt_source_group.add_argument(
         "--input-file",
-        type=argparse.FileType("r"),
+        type=file_or_directory,
         default=None,
         required=False,
-        help="The input file containing the prompts to use for profiling. "
+        help="The input file or directory containing the prompts to use for profiling. "
```

Comment:
I would modify the wording here a bit. Something like
```python
The input file containing the prompts to use for profiling. "
...
For the rankings endpoint-type, a directory should be passed in instead with a "queries.jsonl" file and a "passages.jsonl" file with the same format.
```

I think that would be less messy for the other endpoints and make it clear exactly what you need for the rankings endpoint. If we later expand support for directories for all endpoints, we can think about how to word this help statement differently then.